Lib/test/test_os.py

```diff
@@ -3860,18 +3860,18 @@ def test_oserror_filename(self):
 
         for filenames, func, *func_args in funcs:
             for name in filenames:
-                try:
-                    if isinstance(name, (str, bytes)):
+                if not isinstance(name, (str, bytes)):
```

Comment:
I would move these cases into separate test. It is not anymore about the `filename` attribute of OSError.